client/reader/search-stream/search-follow-button.jsx

```diff
@@ -44,22 +47,53 @@ class SearchFollowButton extends Component {
 				);
 			}
 		}
+		return isPotentialFeedUrl;
+	};
 
-		// If not a potential feed then don't show the follow button
-		if ( ! isPotentialFeedUrl ) {
+	render() {
+		const { query, translate, feeds } = this.props;
+
+		// If the search query hasn't found a feed and the query doesn't look like a feed URL then don't show the follow button
+		if ( ! feeds && ! this.isPotentialFeedUrl( query ) ) {
 			return null;
 		}
 
-		let isFollowing = false;
-		let followTitle = withoutHttp( query );
+		// Check that the query is a URL
+		// Then Loop through feeds and find the feed URL that contains the query
+		// If we find a feed then set the feed object
+		let feed;
+		if ( resemblesUrl( query ) ) {
+			feed = feeds?.find( ( f ) => f.feed_URL.includes( urlToDomainAndPath( query ) ) );
+		}
 
-		if ( feed && feed.name?.length ) {
-			isFollowing = feed.is_following;
-			followTitle = feed.name;
+		// If no feed found, then don't show the follow button
+		if ( ! feed ) {
+			return null;
```

Comment:
In the future it would be good to have a "no results found" state returned by this component. If you search for a non-existent *.wordpress.com blog, the backend returns `https://wordpress.com/blog/feed/` which results in a state like this:

<img width="1102" alt="Screen Shot 2023-06-26 at 4 31 28 pm" src="https://github.com/Automattic/wp-calypso/assets/22446385/cd8322b4-3eb8-4437-b56b-02182999445d">